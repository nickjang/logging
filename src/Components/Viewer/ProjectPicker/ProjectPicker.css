.add-selector-group {
  width: fit-content;
}