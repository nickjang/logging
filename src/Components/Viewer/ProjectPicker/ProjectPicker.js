import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Selector from '../Selector/Selector';
import AddSelector from '../AddSelector/AddSelector';
import SelectorContext from '../../../Context/SelectorContext';
import './ProjectPicker.css';

class ProjectPicker extends Component {
  static contextType = SelectorContext;

  render() {
    const selectorList = (
      this.context.selectors[this.props.project.id].map(selector =>
        <Selector
          key={selector.id}
          id={selector.id}
          type={selector.type}
          calendar={selector.calendar}
          endRange={selector.endRange}
          projectId={this.props.project.id} />
      )
    );

    const selectorTypes =
      [{ type: 'project', label: 'Select the entire project', buttonText: '+ Entire project' },
      { type: 'years', label: 'Make a selection for a year or year range', buttonText: '+ Year(s)' },
      { type: 'months', label: 'Make a selection for a month or month range', buttonText: '+ Month(s)' },
      { type: 'days', label: 'Make a selection for a day or date range', buttonText: '+ Day(s)' }]
        .map(selectorType => {
          return (
            <AddSelector
              key={`${this.props.project.id}-${selectorType.type}`}
              type={selectorType.type}
              label={selectorType.label}
              buttonText={selectorType.buttonText}
              projectId={this.props.project.id} />
          );
        });

    return (
      <li>
        <h3>{this.props.project.title}</h3>
        <fieldset className='add-selector-group group-row'>
          {selectorTypes}
        </fieldset>
        <ul className='selected' role='presentation'>
          {/*on view, combine times and get logs without overlap*/}
          {selectorList}
        </ul >
      </li >
    );
  }
}

ProjectPicker.defaultProps = {
  project: {}
}

ProjectPicker.propTypes = {
  project: PropTypes.object.isRequired
}

export default ProjectPicker;