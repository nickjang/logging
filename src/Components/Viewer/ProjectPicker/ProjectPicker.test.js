import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ProjectPicker from './ProjectPicker';

describe(`ProjectPicker component`, () => {
  it('renders ProjectPicker by default', () => {
    const wrapper = shallow(<ProjectPicker />);
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})