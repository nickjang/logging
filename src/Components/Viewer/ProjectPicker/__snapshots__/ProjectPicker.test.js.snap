// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectPicker component renders ProjectPicker by default 1`] = `
<li
  className="project-picker"
>
  <h3
    className="lg-project-title side-bar-project-title ml-1"
  />
  <fieldset
    className="add-selector-group"
  >
    <div
      className="selector-type-group"
      key="0"
    >
      <AddSelector
        buttonText="+ All logs"
        key="undefined-project"
        label="All logs from this project"
        projectId={null}
        type="project"
      />
      <AddSelector
        buttonText="+ Year(s)"
        key="undefined-years"
        label="Make a selection for a year or year range"
        projectId={null}
        type="years"
      />
    </div>
    <div
      className="selector-type-group"
      key="2"
    >
      <AddSelector
        buttonText="+ Month(s)"
        key="undefined-months"
        label="Make a selection for a month or month range"
        projectId={null}
        type="months"
      />
      <AddSelector
        buttonText="+ Day(s)"
        key="undefined-days"
        label="Make a selection for a day or date range"
        projectId={null}
        type="days"
      />
    </div>
  </fieldset>
  <ul
    className="selected"
    role="presentation"
  />
</li>
`;
